src/cmd_context/cmd_context.h

```diff
@@ -554,6 +562,11 @@ class cmd_context : public progress_callback, public tactic_manager, public ast_
 
     void slow_progress_sample() override;
     void fast_progress_sample() override;
+    smt2::parser* parser() { return m_parser; };
```

Comment:
again, seems api_context is the place to own parsing thunks, cmd_context is also for the shell and does not reset the parser.

Reply:
Thanks for the suggestion. How do I get a reference to the `api_context` inside `Z3_eval_smtlib2_string`?

Reply:
Z3_context is the external pointer name for api_context. The function mk_c(c) coerces it to api_context.
It contains an optional cmd_context pointer (accessed using "cmd()"). It can contain other optional pointers, such as a reusable parser closure.

Reply:
Done.